docs/distribution/README.md

```diff
@@ -0,0 +1,248 @@
+
+## Retrieving Signatures
+
+The list of signatures for an artifact can be retrieved from the registry.
+
+### Option 1: Server-Side Paging
+
+Get a list of paginated signatures from the registry. The response will include an opaque URL that can be followed to obtain the next page of results.
+
+#### Sample Request
+```
+GET http://localhost:5000/v2/hello-world/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/
+```
+
+#### Sample Response
+
+```json
+{
+    "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+    "@nextLink": "{opaqueUrl}",
+    "signatures": [
+        "sha256:1135d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c",
+        "sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b"
+    ]
+}
+```
+
+### Option 2: Client-Side Paging
+
+Get a list of paginated signatures from the registry by specifying the last retrieved item and page size.
+
+#### Sample Request
+```
+GET http://localhost:5000/v2/hello-world/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/?last=sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b&max=10
+```
+
+#### URI Parameters
+
+| Parameter | Description                                                  |
+| --------- | ------------------------------------------------------------ |
+| `last`    | Query parameter for the last item in previous query. Result set will include values lexically after last. |
+| `max`     | Query parameter for max number of items.                     |
+
+#### Sample Response
+
+```json
+{
+    "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+    "signatures": [
+        "sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c",
+        "sha256:1135d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c"
+    ]
+}
+```
+
+## Storing Signatures
+
+Signatures can be stored in the registry as OCI artifacts and linked to target artifacts that they sign.
+
+### Option 1: Client-Driven Signature Linking
+
+The client pushes all artifacts to the registry independently and then links signatures to their targets.
+
+1. Push all artifacts to the registry (for example, using [oras](https://github.com/deislabs/oras)).
+    - Push target artifact `sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`
+    - Push signature artifact `sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c`
+
+2. Link signature artifact to the target it signs.
+
+    #### Sample Request
+    ```
+    PUT https://localhost:6000/v2/hello-world/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c
+    ```
+
+### Option 2: Server-Side Signature Linking
+
+The server detects a signature push and implicitly links the signature to the target.
+#### TODO

```

Comment:
@SteveLasker , we haven't spoken about what the API will look like when the server handles linking of signatures implicitly